pymodbus/transaction/transaction.py

```diff
@@ -153,6 +153,9 @@ async def execute(self, no_response_expected: bool, request: ModbusPDU) -> Modbu
                     return response
                 except asyncio.exceptions.TimeoutError:
                     count_retries += 1
+                except asyncio.exceptions.CancelledError:
+                    self.recv_buffer = b""
```

Comment:
see other PR.

Reply:
Ah, yeah, I guess if we purge the buffer before each send we don't need to clean up on exceptions.